src/java.base/share/classes/sun/util/calendar/CalendarDate.java

```diff
@@ -59,7 +59,8 @@
  * @author Masayoshi Okutsu
  * @since 1.5
  */
-public abstract class CalendarDate implements Cloneable {
+public sealed abstract class CalendarDate implements Cloneable
+        permits BaseCalendar.Date {
```

Comment:
Can we just merge `CalendarDate` and `BaseCalendar.Date` to be the one class?
I think it will greatly simplify the code.

Reply:
`BaseCalendar` is for Gregorian based calendars, so its `Date` class is also represents dates for those calendars, while `CalendarDate` is an abstract class for all calendar implementations. So I don't think merging them is the right direction.